pkg/statistics/handle/syncload/stats_syncload.go

```diff
@@ -231,24 +230,45 @@ func (s *statsSyncLoad) HandleOneTask(sctx sessionctx.Context, lastTask *statsty
 	} else {
 		task = lastTask
 	}
-	return s.handleOneItemTask(sctx, task)
+	resultChan := s.StatsLoad.Singleflight.DoChan(task.Item.Key(), func() (any, error) {
+		return s.handleOneItemTask(sctx, task)
+	})
+	timeout := time.Until(task.ToTimeout)
+	select {
+	case result := <-resultChan:
+		if result.Err == nil {
+			slr := *(result.Val.(*stmtctx.StatsLoadResult))
+			if slr.Error != nil {
+				return task, slr.Error
+			}
+			task.ResultCh <- slr
```

Comment:
How about
```suggestion
			slr := result.Val.(*stmtctx.StatsLoadResult)
			if slr.Error != nil {
				return task, slr.Error
			}
			task.ResultCh <- *slr
```

This can help us to simplify this assessment code.